javascript/ql/src/semmle/javascript/security/dataflow/CleartextLogging.qll

```diff
@@ -42,6 +42,14 @@ module CleartextLogging {
         write.getRhs() = src and
         trg.(DataFlow::SourceNode).flowsTo(write.getBase())
       )
+      or
+      // Taint step through `util.inspect(..)` from Node.js
+      trg = DataFlow::moduleImport("util").getAMethodCall("inspect") and
+      trg.(DataFlow::CallNode).getAnArgument() = src
+      or
+      // Taint step through a `str.replace(..)` call.
+      trg.(DataFlow::MethodCallNode).getCalleeName() = "replace" and
```

Comment:
> In addition to you current dist-compare run, can you try to check the new results on there slugs: https://git.semmle.com/gist/esben/6edfe4176c8b62241c7fa6784b54107a?

Will do.

Reply:
Switching the whole query to flow labels would be great, but probably also a lot of work. It might be easier to factor the standard taint steps in such a way that it is easy to reuse, e.g., only the string manipulation-related ones.

For this configuration, we clearly don't want the standard step from `o` to `o.p`, since we have an additional step from `o.q` to `o` and the combination of these would be bad. Are there any other standard taint steps we _don't_ want?

Reply:
I converted the query to flow-labels, but I didn't change much.  
Flow-labels are for now only used to propagate taint from `process.env` and block other property reads.